File: Backend/Backend/models/Course.js
const mongoose = require('mongoose');

const courseSchema = new mongoose.Schema(
    {
        name: { type: String, required: true },
        description: { type: String, required: true },
        chef: { type: String, required: true }, 
        level: {
            type: String,
            enum: ['Beginner', 'Intermediate', 'Advanced'],
            required: true,
        }, 
        duration: { type: Number, required: true }, 
        date: { type: Date, required: true }, 
        time: { type: String, required: true }, 
        location: { type: String, required: true },
        imageURL: { type: String, required: true }, 
        popularity: { type: Number, default: 0 }, 
    },
    { timestamps: true }
);

module.exports = mongoose.model('Course', courseSchema);

File: Backery/Models/Course.swift
//
//  Course.swift
//  Backery
//
//  Created by Shamam Alkafri on 25/01/2025.
//

import Foundation

struct Course: Codable, Identifiable {
    let id: String
    let name: String
    let description: String
    let chef: String
    let level: String
    let date: String
    let time: String
    let location: String
    let image: String? 
}
